### ISSUE TITLE
Improvement 404 page

### ISSUE THREAD

**Issue by MarkusH-Berlin** [smartstore/Smartstore#541]
We have in Smartstore several 404 pages (check screenshots below)

The first one is great!
The two other ones are terrible, since the customer has no company logo or navigation buttons at all to go on.

Every customer who ends up on these two 404 pages will instantly hit the browser back button, and will be gone forever.

This two 404 pages shows up, if a customer clicks an old .html url for example.

This should be improved, so every customer who comes from wherever with incorrect urls end up on the first of the 404 pages, with company logo and navigation buttons.

 
<img width="1191" alt="Bildschirmfoto 2022-09-07 um 14 35 45" src="https://user-images.githubusercontent.com/112764786/188882003-f58b92d1-e703-4378-90c5-61be067ea2e8.png">
<img width="1458" alt="Bildschirmfoto 2022-09-07 um 14 36 54" src="https://user-images.githubusercontent.com/112764786/188882025-87534342-be22-41cb-a5d9-c624a867dd0e.png">
<img width="1496" alt="Bildschirmfoto 2022-09-07 um 14 37 35" src="https://user-images.githubusercontent.com/112764786/188882029-7509a787-8bfc-4572-8aad-09b4140a268b.png">

**Comment by Michael-Herzog** [comment]
Can you please also provide URLs

**Comment by MarkusH-Berlin** [comment]
Don't need a special URL, I think every smartstore.net reacts like this.

Uncomplete url (= screenshot nr. 1)
https://core.smartstore.com/backend/en/airp

Url with .html at the end, for example links from old orders via smartstore.biz or old .html links that are still available on Google or in forums etc. ( = screenshot nr. 2)
https://core.smartstore.com/backend/en/airp.html

Screenshot number 2 and 3 are caused by the same .html url's (.html at the end)
Your shop gives out screenshot nr. 2, our shop gives out screenshot nr. 3 (this maybe depends on the hoster / server?)

**Comment by Michael-Herzog** [comment]
You should contact our support. They will help you to configure everything correctly.

**Comment by MarkusH-Berlin** [comment]
I told this already the support on phone, they told me to open an issue on GitHub regarding this matter.

This enhancement is not about MY Smartstore, but every Smartstore.

The URLs I posted are from YOUR OWN shop (not my shop)... https://core.smartstore.com/................

**Comment by Michael-Herzog** [comment]
Our support team would never tell you to open an issue here!

**Comment by MarkusH-Berlin** [comment]
Of course they did, from where you think I should know it if not from your support...

I never was on GitHub before.

The Samrtstore support told me to open an issue, in English (I should translate it with google; but I speak English fluently), and add some screenshots.

**Comment by Michael-Herzog** [comment]
I don't know who you've reached here and who has told you this but this is not how our support team works. If there is an issue to be fixed that's reported by our customers we write the issue ourself then we get back to our customers via support ticket.

**Comment by muratcakir** [comment]
> Uncomplete url (= screenshot nr. 1)
https://core.smartstore.com/backend/en/airp

> Url with .html at the end, for example links from old orders via smartstore.biz or old .html links that are still available on Google or in forums etc. ( = screenshot nr. 2)
https://core.smartstore.com/backend/en/airp.html

This is intentional. Static files (like files ending with .html) are handled by `StaticFilesMiddleware` very early in the request pipeline. If the file does not exist **physically**, the respond ends with 404 status code. The server - in your case: IIS - is responsible for showing a status message, which, unfortunately, looks ugly... yes.

To show decently formatted 404 pages even for non-existing static files, `StaticFilesMiddleware` has to be bypassed and the entire request pipeline has to be executed. This is not an option because it can heavily degrade performance and scalability. And we don't want that!

But: you can use the (commercial) `UrlRewriter` plugin to define redirection rules for such non-existing static resources. That's the way to go.

**Comment by MarkusH-Berlin** [comment]
I discovered this issue, when I switched from Smartstore.biz (with .html URLs to Smartstore.net, and I thought; this could be improved, since a customer who end up on that "ugly" 404 page (without company logo and any navigation elements) will probably instantly hit the browser back button and will be gone forever.

I'm already using the URL Rewriter Plugin successfully, and I covered maybe 95% of our old .html URLs. That works perfect for me.
